spec/lib/facilities/gis_client_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Facilities::GisClient do
+  RSpec::Matchers.define :has_order_and_offset do |order, offset|
+    match do |params|
+      params[:orderByFields].eql?(order) && params[:resultOffset].eql?(offset)
+    end
+  end
+
+  describe 'get_all_facilities' do
+    subject { described_class.new }
+
+    it 'passes the correct offset to the query when looping twice' do
+      r1 = double(Faraday::Response.new)
+      r2 = double(Faraday::Response.new)
+
+      allow_any_instance_of(Faraday::Connection)
```

Comment:
You can declare `allow_any_instance_of(Faraday::Connection)` for each offset just once if you move them into a `before(:each)` block.